### ISSUE TITLE
Update ghostscript

### ISSUE THREAD

**Issue by Erick555** [flathub/org.gimp.GIMP#29]
Ghostscript is source of endless critical security vulnerabilities which allows for RCE:
https://bugs.chromium.org/p/project-zero/issues/detail?id=1640
https://www.kb.cert.org/vuls/id/332928

The version [used here](https://github.com/flathub/org.gimp.GIMP/blob/master/org.gimp.GIMP.base-modules.json#L284) is [2 years old](https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/tag/gs920).

I would strongly consider removing ghostscript from this package. At minimum update ghostscript to latest release (9.23 - but it's still vulnerable to publicly available exploits.)

**Comment by Jehan** [comment]
I will update the version packaged in our flatpak.

As for removing it altogether, please ask directly in our upstream bug tracker: https://gitlab.gnome.org/GNOME/gimp/

Obviously your request will have more weight if it comes together with proposition of what to replace it with in order not to lose a feature (postscript support). And even better, a patch would be really appreciated.

**Comment by Erick555** [comment]
The problem is that when installing gimp from distro repo, i.e. in debian, ubuntu ,archlinux - ghostscript is an optional dependency - users don't have to install it. In flatpak there is no choice - ghostscript will be always installed. This combined with [full filesystem access](https://github.com/flathub/org.gimp.GIMP/blob/master/org.gimp.GIMP.json#L13) is unfortunate combination.

In the end it's all about choice - either lose a feature or lose security. I would prefer the former but if you choose the latter - so be it.

Nevertheless updating ghostscript is the right thing to do anyway.

**Comment by Jehan** [comment]
> In the end it's all about choice - either lose a feature or lose security. I would prefer the former but if you choose the latter - so be it.

I haven't "chosen" anything yet, and am certainly not the only developer in GIMP. Others have their words to say as well (and in particular the maintainer can have the finale word).
Once again, if really you wish to see ghostscript disappear, you should move the discussion on our main bugtracker. Nobody else from the GIMP team read this one but me (it exists only because flathub make us have this github repo :-/), I think. Keep only this report for the "updating ghostscript in flatpak" part.

In any case, my opinion is that postscript is common enough for us to want to have support. Now I would assume/hope other FLOSS libraries exist for it as alternative (this won't be the first time we drop a library for another better maintained). Even if there were no alternatives at all, we may even drop the support if really the arguments were compelling (i.e. if the security risks are just too high; I haven't really taken the time to read your links in details, so I don't know).

I really don't know, that's all just rambling on my side right now. The point is that you should move this discussion if really you care about it. I am not going to make any rash decision right here, right now, and alone without the others.

**Comment by Jehan** [comment]
> I'm not advocating for upstream gimp to drop ghostscript. I'm advocating for dropping ghostscript from downstream gimp package in flathub.

I am indeed the maintainer of this package, and I am also a core GIMP developer. This is the official flatpak package maintained by the GIMP team, and therefore is as upstream and core as it can get.
This package is meant to be a full-featured GIMP (as much as possible; some features were lost because of the sandboxing, other things, we are working on, etc.).

So either we decide that ghostscript is too bad and should not be in GIMP, or we don't. This package does not take separate decisions.

> Even if you don't want or can't drop ghostscript then please at least update it.

I already agreed on this. I will do so as soon as I can make the time. :-)

**Comment by nedrichards** [comment]
@Jehan fwiw we can remove the issue tracker here if you want and you can direct people directly upstream to report bugs in the readme.

**Comment by Jehan** [comment]
@nedrichards Yeah actually it would be nice to have all the reports in a single place.
I assume though that pull requests will stay here?

**Comment by Jehan** [comment]
> On the other hand if you don't see any value in reporting issues like this one then indeed it doesn't make sense to host issue tracker here.

I don't understand why you keep on insisting I don't want your bug reports. I already told you both the issues you reported were perfectly valid! The first one, updating ghostscript, I already agreed on it, and said I would do it.

The second issue, I said it was at least worth considering and investigating. Of course it is not as easy as updating. We are talking about potentially removing a very important feature (postscript is a very common format) so the security risk would need to be quite important (like being able to run arbitrary commands from a malevolent postscript file, or something). Actually we would probably look for alternative libraries first so that we can keep the feature while dropping ghostscript.
And in any case, **I won't take such a decision here alone without discussing with others** (you could imagine that many people would be quite displeased if I decided to drop postscript support on a whim)!
This is why I said this needs to be on the main bugtracker. So please, let's make it the last I say it: I am happy to see your issue reports and I see value in the one you made. Thanks for them.

@nedrichards The discussion above just makes it obvious that we need to have a single place for handling our bugs. I can't have people asking me for core GIMP issues here (and telling me I don't want their reports if I ask them to move the report there).
What's the process? Also I want to keep the bug reports already opened here. Should I copy them manually?

**Comment by Jehan** [comment]
> you totally misinterpreted what I wrote above.

Ok. Sorry if I did.

> I responded to your request for closing issue tracker on github which means you won't hear anything from me anymore and perhaps other flatpak users.

Most flatpak users are already directly reporting all their issues to the main bugtracker. We actually have very very few reports on this tracker. So I don't worry. :-)
Also that's GNOME bugtracker, so most people who report bugs to FLOSS already have an account anyway.

**Comment by nedrichards** [comment]
@Jehan yes, PRs stay here since they apply to the flathub project. 

I fear you'll have to manually move the issues as you say, then add the readme pointing people to the upstream gitlab and make sure the appdata has the bug tracker URL as well. After that we can disable the issues module for this repo.

**Comment by Erick555** [comment]
@Jehan You forgot to update ghostscript version in some places:
https://github.com/flathub/org.gimp.GIMP/blob/fa4495a97aff355acf91ad454930abb030e4dea0/org.gimp.GIMP.json#L28
https://github.com/flathub/org.gimp.GIMP/blob/fa4495a97aff355acf91ad454930abb030e4dea0/org.gimp.GIMP.WithBase.json#L38
https://github.com/flathub/org.gimp.GIMP/blob/32af3bbb8df435c971c88a9704554c89501acf48/org.gimp.GIMP.app-modules.json#L74

Also you have to change `lcms2` to `lcms2mt`  here: https://github.com/flathub/org.gimp.GIMP/blob/master/org.gimp.GIMP.base-modules.json#L299 . See https://github.com/flathub/org.inkscape.Inkscape/blob/master/org.inkscape.Inkscape.json#L261 for example.

**Comment by Jehan** [comment]
Updated.

```
commit d85d33a0e02689a4131f4f05589456eeb17f018d
Author: Jehan <jehan@girinstud.io>
Date:   Mon Sep 24 16:31:53 2018 +0200

    Issue #29:  Update ghostscript.

 org.gimp.GIMP.base-modules.json | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

commit 6da9d5860209b5df53083008da43333db645f6cb
Author: Jehan <jehan@girinstud.io>
Date:   Mon Sep 24 18:55:35 2018 +0200

    Fix ghostscript 9.25 build.
    
    lcms2/ folder was apparently renamed to lcms2mt/.
    This should fix this error when building the ghostscript module:
    > cp: target 'lcms2/' is not a directory

 org.gimp.GIMP.base-modules.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

commit 5a3b2d47c156010b5c1f7679743e6c001b50f112 (HEAD -> master, origin/master, origin/HEAD)
Author: Jehan <jehan@girinstud.io>
Date:   Tue Sep 25 00:00:34 2018 +0200

    Forgot some places to cleanup ghostscript side files.
    
    Thanks to Erick555 for the notice.

 org.gimp.GIMP.WithBase.json    | 2 +-
 org.gimp.GIMP.app-modules.json | 2 +-
 org.gimp.GIMP.json             | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)
```

Build successful: https://flathub.org/builds/#/builders/2/builds/4730

**Comment by Erick555** [comment]
Thank you!
